### PR TITLE
[GUI] Get images from the current GUI using gui.get_image()

### PR DESCRIPTION
**Pull request by Rullec**
Related issue = From Wechat

Our classmate in GAMES201 said that, he want to fetch the current image array from the GUI. 

So here is a trivial implmention, please feel free to give me more suggestions.

Btw, do we need more test cases about this func?

### CODE REVIEW

File: python/taichi/misc/gui.py [hunk 1]
```diff
@@ -75,6 +75,10 @@ def cook_image(self, img):
         assert res == self.res, "Image resolution does not match GUI resolution"
         return np.ascontiguousarray(img)
 
+    def get_image(self):
+        self.core.get_img(self.img.ctypes.data)
```
Comment (archibate) [inline python/taichi/misc/gui.py:79]: What if self.img is not a contigugous array? We may want to enforce this in set_image.

File: docs/gui.rst [hunk 2]
```diff
@@ -313,6 +313,12 @@ A *event filter* is a list combined of *key*, *type* and *(type, key)* tuple, e.
 Image I/O
 ---------
 
+.. function:: gui.get_image()
+
+    :return a ``np.ndarray`` which is the current image shown on the GUI.
```
Comment (archibate) [inline docs/gui.rst:318]: Can we emphersize that the returned array is a RGBA array?

Comment (Rullec) [reply docs/gui.rst:318]: Yes, good.

RGBA is guranteed by the type of `canvas::img` `Array2d<Vector4d>`, right?

File: python/taichi/misc/gui.py (hunk 1 above)
Comment (Rullec) [reply python/taichi/misc/gui.py:79]: Agree, WIP

**Comment by Rullec** [comment]
How can I trigger the `Check PR Title` mannually after revising?

**Comment by archibate** [comment]
> How can I trigger the `Check PR Title` mannually after revising?

Just update the doc with RGBA, and it wil be triggered again.

File: python/taichi/misc/gui.py (hunk 1 above)
Comment (archibate) [reply python/taichi/misc/gui.py:80]: Line 115:
```py
            self.img = self.cook_image(img)
```
Could you check if `cook_image` return a RGBA array? I didn't remember..

Comment (Rullec) [reply python/taichi/misc/gui.py:80]: Just now I double check it. If the input of `cook_image` is a valid image (no matter graylike or rgb or rgba), the return will be an RGBA array.

The only exception is when `img`'shape is [x, y, 2], in this case our `cook_image` will fail to return RGBA array. But [x, y, 2] is an invalid image size, IMO.

**Comment by archibate** [comment]
Thanks! Merging this for make a rush for v0.6.17 (#1436).

**Pull request merged**